qtism/data/storage/xml/marshalling/Html5ContentMarshaller.php

```diff
@@ -0,0 +1,117 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+namespace qtism\data\storage\xml\marshalling;
+
+use DOMElement;
+use qtism\common\utils\Version;
+use qtism\data\content\BodyElement;
+use qtism\data\content\enums\Role;
+use qtism\data\content\FlowCollection;
+use qtism\data\content\xhtml\html5\Html5Element;
+use qtism\data\QtiComponent;
+use qtism\data\QtiComponentCollection;
+use qtism\data\storage\xml\versions\QtiVersion;
+
+/**
+ * The Marshaller implementation for object elements of the content model.
+ */
+class Html5ContentMarshaller extends ContentMarshaller
+{
+    /**
+     * @param DOMElement $element
+     * @param QtiComponentCollection $children
+     * @return mixed
+     * @throws UnmarshallingException
+     */
+    protected function unmarshallChildrenKnown(DOMElement $element, QtiComponentCollection $children): QtiComponent
+    {
+        $fqClass = $this->lookupClass($element);
+        $component = new $fqClass();
+        $component->setContent(new FlowCollection($children->getArrayCopy()));
+
+        if (($xmlBase = self::getXmlBase($element)) !== false) {
+            $component->setXmlBase($xmlBase);
+        }
+
+        $this->fillBodyElement($component, $element);
+
+        return $component;
+    }
+
+    /**
+     * @param QtiComponent $component
+     * @param array $elements
+     * @return DOMElement
+     */
+    protected function marshallChildrenKnown(QtiComponent $component, array $elements): DOMElement
+    {
+        /** @var Html5Element $component */
+        $prefix = $component->getTargetNamespacePrefix();
+        $version = QtiVersion::create($this->getVersion());
+        $namespace = $version->getExternalNamespace($prefix);
+
+        $element = static::getDOMCradle()->createElementNS(
+            $namespace,
+            $prefix . ':' . $component->getQtiClassName()
+        );
+
+        if ($component->hasXmlBase() === true) {
+            self::setXmlBase($element, $component->getXmlBase());
+        }
+
+        foreach ($elements as $e) {
+            $element->appendChild($e);
+        }
+
+        $this->fillElement($element, $component);
+
+        if ($component->hasTitle()) {
+            $this->setDOMElementAttribute($element, 'title', $component->getTitle());
+        }
+
+        if ($component->hasRole()) {
+            $this->setDOMElementAttribute($element, 'role', Role::getNameByConstant($component->getRole()));
+        }
+
+        return $element;
+    }
+
+    protected function setLookupClasses()
+    {
+        $this->lookupClasses = [
+            "qtism\\data\\content\\xhtml",
+            "qtism\\data\\content\\xhtml\\html5",
+            "qtism\\data\\content\\xhtml\\text",
+        ];
+    }
+
+    protected function fillBodyElement(BodyElement $bodyElement, DOMElement $element): void {
```

Comment:
could fix the indentation?
I'd recommend you apply PSR12 on your files.

Reply:
fixed